drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -722,6 +722,23 @@ static int nrf5_stop(const struct device *dev)
 	return 0;
 }
 
+#if NRF_802154_CARRIER_FUNCTIONS_ENABLED
```

Comment:
I think this should get a Kconfig entry in `modules/hal_nordic/Kconfig`.

Plus, a change in a specific radio driver, also merits a separate commit.

Reply:
Indeed we should. Too bad I wasn't CC'ed on that.

Reply:
Moved change to separate commit